http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Brotli4JHttpIT.java

```diff
@@ -0,0 +1,160 @@
+                .contentType(MediaType.TEXT_PLAIN)
+                .get("/compression/text")
+                .then()
+                .statusCode(200)
+                .header(HttpHeaders.CONTENT_LENGTH, String.valueOf(CONTENT_LENGTH_DEFAULT_TEXT_PLAIN))
+                .header(HttpHeaders.CONTENT_ENCODING, nullValue())
+                .body(is(DEFAULT_TEXT_PLAIN))
+                .log().all();
+    }
+
+    @Test
+    public void checkJsonBrotli4JCompression() throws IOException {
+        int originalJsonLength = calculateOriginalJsonLength("/json_sample.json");
+        assertBrotli4JCompression("/compression/brotli/json", MediaType.APPLICATION_JSON, BROTLI_ENCODING, BROTLI_ENCODING, originalJsonLength );
+    }
+
+    @Test
+    public void checkCompressedAndDecompressedWithQuarkus() {
+        testCompressedAndDecompressed("/compression/text", DEFAULT_TEXT_PLAIN);
+    }
+    @Test
+    public void checkCompressedAndDecompressedJSONWithQuarkus() {
+        testCompressedAndDecompressed("/compression/brotli/json", JSON_CONTENT);
+    }
+
+    @Test
+    public void checkCompressedAndDecompressedXMLWithQuarkus() {
+        testCompressedAndDecompressed("/compression/brotli/xml", "Bob Dylan");
+    }
+
+    @Test
+    public void checkXmlBrotli4JCompression() throws IOException {
+        int originalXMLLength = calculateXmlLength();
+        assertBrotli4JCompression("/compression/brotli/xml", MediaType.APPLICATION_XML, BROTLI_ENCODING, BROTLI_ENCODING, originalXMLLength);
+    }
+
+    private void testCompressedAndDecompressed(String compressionPath, String contentExpected) {
+        Response response = app.given()
+                .header(HttpHeaders.ACCEPT_ENCODING, BROTLI_ENCODING)
+                .get(compressionPath)
+                .then()
+                .statusCode(200)
+                .header(HttpHeaders.CONTENT_ENCODING, BROTLI_ENCODING)
+                .log().all()
+                .extract().response();
+        byte[] compressedBytes = response.asByteArray();
+
+        Response decompressionResponse = app.given()
+                .header(HttpHeaders.CONTENT_ENCODING, BROTLI_ENCODING)
+                .body(compressedBytes)
+                .post("/compression/decompression")
+                .then()
+                .statusCode(200)
+                .log().all()
+                .extract().response();
+        assertThat(decompressionResponse.asString(), containsString(contentExpected));
+    }
+
+    private int calculateOriginalJsonLength(String jsonFilePath) throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream(jsonFilePath)) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(inputStream);
+            String jsonString = objectMapper.writeValueAsString(jsonNode);
+            return jsonString.getBytes().length;
+        }
+    }
+
+    private int calculateXmlLength() throws IOException {
```

Comment:
Wouldn't be better have the xml file path as parameter same as in `calculateOriginalJsonLength` method?